tests/src/test/java/alluxio/job/transform/format/parquet/ParquetWriterIntegrationTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.job.transform.format.parquet;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.AlluxioURI;
+import alluxio.job.JobIntegrationTest;
+import alluxio.job.transform.format.TableSchema;
+import alluxio.uri.Authority;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Unit tests for {@link ParquetWriter}.
+ */
+public class ParquetWriterIntegrationTest extends JobIntegrationTest {
+  @Test
+  public void createLocalFileWithNonExistingParents() throws IOException {
+    Path path = Files.createTempFile("local", "parquet");
+    Files.delete(path);
+    AlluxioURI uri = new AlluxioURI("file:///" + path.toString());
+
+    assertFalse(Files.exists(path));
+    TableSchema schema = new ParquetSchema(ParquetTestUtils.SCHEMA);
+    ParquetWriter writer = ParquetWriter.create(schema, uri);
+    writer.close();
+    assertTrue(Files.exists(path));
+  }
+
+  @Test
+  public void createAlluxioFileWithNonExistingParents() throws Exception {
```

Comment:
could you also add a test where the uri does not have scheme/authority?   would it pick the correct alluxio master?

Reply:
I don't think so.
If there is no scheme, then the parquet-mr will treat it as a local filesystem path.
If there is no authority, then the parquet-mr library will try to connect to fs.defaultFs

Reply:
I added precondition checks for the URI to enforce the assumptions

Reply:
In the future, we might only store the path if it refers to a path on the same Alluxio file system. So the hostname of the Alluxio master can change. 
In that case, transform jobs might need to handle a uri without scheme/authority...

Reply:
but this change can probably wait..

Reply:
if there no scheme/authority, then the master address can be determined through `ServerConfiguration`?